refinery/cli.py

```diff
@@ -16,7 +17,32 @@ def start(cur_dir: str):
         cur_dir (str): The current directory.
     """
 
-    def _start_server():
+    def _start_server(check_for_update: bool):
+
+        if check_for_update:
+            repo = Repo(search_parent_directories=True)
+            sha_local = repo.head.object.hexsha
+
+            # https://stackoverflow.com/questions/62525382/how-to-get-the-latest-commit-hash-on-remote-using-gitpython
+            repo_url = f"{REFINERY_REPO}.git"
+            process = subprocess.Popen(
+                ["git", "ls-remote", repo_url], stdout=subprocess.PIPE
+            )
+            stdout, stderr = process.communicate()
+            sha_remote = re.split(r"\t+", stdout.decode("ascii"))[0]
+
+            if sha_local != sha_remote:
+                msg.info(
+                    "A new version of refinery is available. Should this be pulled? (y/n)"
+                )
+                user_input = input("> ")
+                if user_input == "y":
+                    repo_origin = repo.remotes.origin
+                    repo_origin.pull()
+                    msg.good(f"refinery has been updated to commit {sha_remote}.")
```

Comment:
not sure if gitpython work with it but a pull can run into some issues e.g. if you have a fork and uncommitted changes -> maybe we can check it?

Reply:
I have found a solution and implemented it :)